### ISSUE TITLE
label-content-name-mismatch does not account for formatting markup

### ISSUE THREAD

**Issue by curtbellew** [dequelabs/axe-core#3316]
Rule label-content-name-mismatch doesn't seem to account for formatting markup.  Example:

```html
<button aria-label="button label">button<br>label</button>
```

Will return a violation.

Product: axe-core

Expectation: Formatting should be disregarded.

Actual: The string is compared directly so any HTML tabs present cause a violation

Motivation: 

<pre><code>
axe-core version: 4.1.3
axe-webdriver, extension or other integration version: X.Y.Z

Browser and Assistive Technology versions

For Tooling issues:
- Node version: XX  
- Platform:  
</code></pre>

**Comment by WilcoFiers** [comment]
Thank you for reporting this. I've confirmed the issue. I agree, this shouldn't be failing. We'll look at what we can do about this.

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core dev branch code base,
for the test script '<button aria-label="button label">button<br>label</button>' , where label content is formatted markup, should not encounter the failure on `label-content-name-mismatch` rule.
Not seeing such failure, can find in screenshot
![image](https://user-images.githubusercontent.com/47409579/151569180-3c0f6f32-9827-4f91-8334-80d33d855658.png)
